Operators/Types/lib/string/IndexOf.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections;
+using System.Text.RegularExpressions;
+using T3.Core.Logging;
+using T3.Core.Operator;
+using T3.Core.Operator.Attributes;
+using T3.Core.Operator.Slots;
+
+namespace T3.Operators.Types.Id_fc0a5e68_9915_4323_b2a4_2491fa5d59a9
+{
+    public class IndexOf : Instance<IndexOf>
+    {
+        [Input(Guid = "841784c4-0ca7-41cd-8d79-bbe4989e0842")]
+        public readonly InputSlot<string> OriginalString = new();
+
+        [Input(Guid = "81a7aa30-eab9-4637-bb11-7c0940460afb")]
+        public readonly InputSlot<string> SearchPattern = new();
+
+        public IndexOf()
+        {
+            Index.UpdateAction = Update;
+        }
+
+        private void Update(EvaluationContext context)
+        {
+            string searchPattern = SearchPattern.GetValue(context);
```

Comment:
```suggestion
            string searchPattern = SearchPattern.GetValue(context);
            string originalString = OriginalString.GetValue(context);
            if (string.IsNullOrEmpty(searchPattern))
            {
                Index.Value = -1;
                return;
            }

            Index.Value = originalString.IndexOf(searchPattern);
```

Although the code looks valid and clean there is a small but important detail hidden here:

If you exit early, before getting the value for `OriginalString.GetValue(context);` this input will always stay dirty. This means that every Operator right of it will stay dirty and can't be cached.

For that reason, gathering all inputs before existing is the better thing to do here. 

Another detail: With string operators I'm personally never sure when it throws exceptions. In your case, you should also check `originalString` for null before call `IndexOf`. Otherwise this operator will throw an exception if connected to a null string. (Currently by design throwing and exception will exist the exception - which looks like a crash to most users).

If in doubt add a try/catch statement around the functions in question, just to be on the save side.

Reply:
I see, interesting! Thanks for the explanation. I've added the suggestions.